### PR TITLE
Refactor design of Queues wizard page

### PR DESCRIPTION
**Pull request by tmscarla**
## Description
This PR refactors the design of Queues wizard page. Now each queue has its own dedicated container, and compute resources are not wrapped in an additional container.

Furthemore:
* `Add queue` button has been moved to the top-right part of the container
* Compute resources names are now displayed in the wizard
* Added Queue name validation

## How Has This Been Tested?

* Manual tests on local environment
* Unit tests

## References

* https://cloudscape.design/components/column-layout/

## Screenshots

Screenshot 2023-03-02 at 14 25 18

## PR Quality Checklist

- [x] I added tests to new or existing code
- [x] I removed hardcoded strings and used [`react-i18next`](https://react.i18next.com/) library ([useTranslation hook](https://react.i18next.com/latest/usetranslation-hook) and/or [Trans component](https://react.i18next.com/latest/trans-component)), see an example [here](https://github.com/aws/aws-parallelcluster-ui/commit/a6f1e2aa46b245b5bf7500a04b83195477a5cfa5)
- [ ] I made sure no sensitive info gets logged at any time in the codebase (see [here](https://cheatsheetseries.owasp.org/cheatsheets/Logging_Cheat_Sheet.html)) (e.g. no user info or details, no stacktraces, etc.)
- [ ] I checked that infrastructure/update_infrastructure.sh runs without any error
- [x] I checked that `npm run build` builds without any error
- [x] I checked that clusters are listed correctly
- [x] I checked that a new cluster can be created (config is produced and dry run passes)
- [ ] I checked that login and logout work as expected

In order to increase the likelihood of your contribution being accepted, please make sure you have read both the [Contributing Guidelines](../CONTRIBUTING.md) and the [Project Guidelines](../PROJECT_GUIDELINES.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by mendaomn** [comment]
General remark: how is this PR going to affect the e2e tests?

File: frontend/src/old-pages/Configure/Queues/Queues.tsx [hunk 1]
```diff
@@ -108,7 +112,16 @@ function queueValidate(queueIndex: any) {
   const queueName = getState([...queuesPath, queueIndex, 'Name'])
   const [queueNameValid, error] = validateQueueName(queueName)
   if (!queueNameValid) {
-    setState([...errorsPath, 'name'], i18next.t(queueNameErrorsMapping[error!]))
+    let errorMessage: string
+    if (error === 'max_length') {
+      errorMessage = i18next.t(queueNameErrorsMapping[error], {
+        maxChars: QUEUE_NAME_MAX_LENGTH,
+      })
+      console.log(errorMessage)
```
[the remark this answers is not in the document]
Comment (tmscarla) [reply frontend/src/old-pages/Configure/Queues/Queues.tsx:120]: yes, will remove in next revision

File: frontend/src/old-pages/Configure/Queues/Queues.tsx [hunk 2]
```diff
@@ -406,137 +431,174 @@ function Queue({index}: any) {
     [index],
   )
 
+  const adapter = useComputeResourceAdapter()
```
[the remark this answers is not in the document]
Comment (tmscarla) [reply frontend/src/old-pages/Configure/Queues/Queues.tsx:434]: nice catch!

File: frontend/src/old-pages/Configure/Queues/Queues.tsx [hunk 3]
```diff
@@ -406,137 +431,174 @@ function Queue({index}: any) {
...
+                </SpaceBetween>
+              }
+            >
+              {queue.Name}
+            </Header>
+          }
+        >
+          <ColumnLayout borders="horizontal">
+            <Box>
+              <Box margin={{bottom: 'xs'}}>
```
[the remark this answers is not in the document]
Comment (tmscarla) [reply frontend/src/old-pages/Configure/Queues/Queues.tsx:482]: Gotcha!

File: frontend/src/old-pages/Configure/Queues/MultiInstanceComputeResource.tsx [hunk 4]
```diff
@@ -201,9 +201,18 @@ export function ComputeResource({
   return (
     <div className="compute-resource">
       <div style={{display: 'flex', flexDirection: 'column', gap: '10px'}}>
-        <Box margin={{top: 'xs'}} textAlign="right">
-          {index > 0 && <Button onClick={remove}>Remove Resource</Button>}
-        </Box>
+        <ColumnLayout columns={2}>
+          <h4>{computeResource.Name}</h4>
```
[the remark this answers is not in the document]
Comment (tmscarla) [reply frontend/src/old-pages/Configure/Queues/MultiInstanceComputeResource.tsx:205]: I tried with `Box` but I obtained a worse result imho. Headings give enough vertical space between queue name and compute resource fields.

File: frontend/locales/en/strings.json [hunk 5]
```diff
@@ -807,10 +809,6 @@
         "scriptWithArgs": "You must specify a script path if you specify args.",
         "customAmiSelect": "You must select an AMI ID if you use a custom AMI."
       },
-      "container": {
-        "title": "Queues",
-        "description": "Configure up to {{limit}} queues (Slurm partitions)."
-      },
       "slurmMemorySettings": {
```
[the remark this answers is not in the document]
Comment (tmscarla) [reply frontend/locales/en/strings.json:812]: According to Figma, yes.

File: frontend/src/old-pages/Configure/Queues/Queues.tsx [hunk 6]
```diff
@@ -256,27 +269,49 @@ function queuesValidate() {
...
+
   return (
-    <Container>
+    <>
       <Header
         variant="h3"
-        description={t('wizard.queues.computeResource.header.description', {
-          limit: MAX_COMPUTE_RESOURCES,
-        })}
+        actions={
```
[the remark this answers is not in the document]
Comment (tmscarla) [reply frontend/src/old-pages/Configure/Queues/Queues.tsx:289]: According to Figma, yes.

**Review by mendaomn — APPROVED** [review]

**Comment by tmscarla** [comment]
> General remark: how is this PR going to affect the e2e tests?

Should not affect e2e since in the `Queue` page the only element we look for is the heading, which is untouched. Tested locally, it works.

**Pull request merged**